Sources/Event Logging/TracksService.m

```diff
@@ -243,7 +245,8 @@ - (void)switchToAnonymousUserWithAnonymousID:(NSString *)anonymousID
     self.token = nil;
 
     #if TARGET_OS_IPHONE
-    [ExPlat configureWithPlatform:_eventNamePrefix oAuthToken:nil userAgent:self.userAgent anonId:anonymousID];
+    NSString *platform = _platform ? _platform : _eventNamePrefix;
```

Comment:
This `platform` conditional assignment logic is duplicated at line 233 above.

Have you considered extracting it in an helper method? `[self platformForExPlat]`?

Alternatively, since it seem like `[ExPlat configureWithPlatform:oAuthToken:userAgent:anonId:]` is the only consumer of this computed `platform` value, what do you think of moving the computation in there?

Reply:
This workaround seems critical for the correct attribution of events and experiments but there are no tests for it.

Reply:
>This workaround seems critical for the correct attribution of events and experiments but there are no tests for it.

Resolved in 2f694f46df1052427097dc025a366fdf73e7147d

Reply:
> Have you considered extracting it in an helper method? [self platformForExPlat]?

Resolved in d15c2da9c316dead254cb1c9c6caf9638bb23e96

Reply:
> This workaround seems critical for the correct attribution of events...

The addition of the `platform`  property only impacts **ExPlat**, Track events are still using the `eventNamePrefix` property.